Mathlib/NumberTheory/Zsqrtd/GaussianInt.lean

```diff
@@ -50,6 +50,7 @@ def GaussianInt : Type :=
   Zsqrtd (-1)
 #align gaussian_int GaussianInt
 
+@[inherit_doc]
```

Comment:
Should we really be putting this on `local` notation?

Reply:
I've filed #11942 with more such changes :-) I'm doing this for ~two reasons:
0. Why not, does it hurt?
0'. These changes remove some nolint entries, so something somewhere cares about this.
1. If I encounter some local notation in the middle of a file, that might not be obvious to me --- so showing some information hover seems useful.
As this is *local* notation, there is no sense in updating the referenced declaration's docstring. (If there is, I'd argue this notation should be scoped, but not local.) Hence, a bare inherit_doc feels appropriate.

I'm happy to move these changes to the other PR if you prefer.